pkg/datapath/linux/datapath.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linux
+
+import (
+	"github.com/cilium/cilium/pkg/datapath"
+)
+
+// DatapathConfiguration is the static configuration of the datapath. The
+// configuration cannot change throughout the lifetime of a datapath object.
+type DatapathConfiguration struct {
+	// HostDevice is the name of the device to be used to access the host.
+	HostDevice string
```

Comment:
If you want to ensure that this will never be changed throughout the lifetime of the datapath object, you could make it private and have setters that return an error if it is tried to be changed after being set (i.e., the value of the field is its zero value, in this case "").

Reply:
DatapathConfiguration  is passed in as a copy into NewDatapath so it is already impossible to ever change this